data_collection/gazette/spiders/base.py

```diff
@@ -188,3 +188,63 @@ def get_gazette(self, document):
             file_urls=(document[0],),
             power="executive",
         )
+
+
+class ImprensaOficialSpider(BaseGazetteSpider):
+    @staticmethod
+    def _get_next_month(current_date):
+        month = (current_date.month % 12) + 1
+        year = current_date.year + (current_date.month + 1 > 12)
+        return date(year, month, 1)
+
+    @staticmethod
+    def _extract_public_url(file_url):
+        if not file_url:
+            return
+        # original: http://www.imprensaoficial.org/pdf/baixar.php?
+        #           arquivo=../pub/prefeituras/ba/ameliarodrigues/2020/proprio/1582.pdf
+        # download: http://www.imprensaoficial.org/pub/
+        #           prefeituras/ba/ameliarodrigues/2020/proprio/1582.pdf
+        index = file_url.find("arquivo=../") + 11
```

Comment:
**suggestion**: Usually the use of [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)) is an anti-pattern as it makes difficult for other developers to understand the reasoning of the choice of that number (`11` in this case).

In this particular case, I believe it would be clearer if we use a regex directly from the selector that returns the desired URL (and remove the need for a method just for this handling). Something like:

```
file_url = response.css("div.entry-content a::attr(href)").re_first(r"arquivo=\.\.\/(.*)")
```

To get the absolute URL, we can use `urljoin` from the response:

```
file_url = response.urljoin(
    response.css("div.entry-content a::attr(href)").re_first(r"arquivo=\.\.\/(.*)")
)
```